Makefile

```diff
@@ -1,14 +1,10 @@
 .PHONY: all
 all: test build
 
-internal/agent/agent.pb.go: proto/agent.proto
```

Comment:
I'd like to keep the dependency on the source `proto/agent.proto`

Reply:
Part of the PR was moving this to use `go generate` rather than the makefile. It is... kind of messy as go generate is kind of messy.

Reply:
The same reason here, I'd like Make to recognize the change, and run `go generate`